commitment/mock.go

```diff
@@ -64,3 +65,63 @@ func ParseTapscriptSibling(t testing.TB, ts string) *TapscriptPreimage {
 
 	return siblingHex
 }
+
+func NewTestFromProof(t testing.TB, p *Proof) *TestProof {
+	t.Helper()
+
+	tp := &TestProof{
+		TaprootAssetProof: &TestTaprootAssetProof{
+			Proof:   mssmt.HexProof(t, &p.TaprootAssetProof.Proof),
+			Version: uint8(p.TaprootAssetProof.Version),
+		},
+	}
+	if p.AssetProof != nil {
+		tp.AssetProof = &TestAssetProof{
+			Proof:   mssmt.HexProof(t, &p.AssetProof.Proof),
+			Version: uint8(p.AssetProof.Version),
+			AssetID: hex.EncodeToString(p.AssetProof.AssetID[:]),
+		}
+	}
+
+	return tp
+}
+
+type TestProof struct {
+	AssetProof        *TestAssetProof        `json:"asset_proof"`
```

Comment:
Maybe prefix with Smt? IIUC this is the smt inclusion proof for na asset while the bottom is the entire state transition proof?

Reply:
I tried to keep the names of the original structs:
https://github.com/lightninglabs/taproot-assets/blob/0a58cdbf873f4634f05aa6f40fa75f888f0f5195/commitment/proof.go#L54

Though I agree it's not super clear now. IMO it was way clearer what the proofs meant when they were called `AssetProof` and `TaroProof`, as that directly corresponded to the two levels of trees.

Happy to rename either of them, but I would suggest to do it in both the test struct and the "production code" struct?